---
title: "Genomics Workshop: Task 1"
author: 
  Ruining Dong^[dong.rn@wehi.edu.au], 
  James Fu, Daniel Cameron
output: 
  rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Genomic Workshop Task 1}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
```

## Sequence alignment

The first step in a variant calling workflow is to align the reads to a reference genome.
Typical human whole genome sequencing (WGS) data contains hundreds of millions or billions of reads.
These data sets take many hours to align so for this workshop we are going to align a very small subset of real sequencing data.

The data we are using today is located in the colo829 directory.

Sequence alignment tools are command-line tools so for this first task, we are going to use the command-line terminal.

To start, let us navigate to the directory containing our data.

```{bash}
cd colo829
ls
```

The `cd` command is the "change directory" command.
It changes our working directory from our "home" directory to the colo829 directory.

The `ls` command is the "list" command.
It outputs all the files and directories in the current directory.

Our input data is paired-end sequencing data for a tumour sample and a matched normal.

This `fastq` files are output from the sequencing machine and we need to align them to the reference genome.
In this workshop, we are going to use chromosome 7 of the hg19 human reference genome as our reference.

We will use `bwa mem` for our sequence alignment. If we type:

```{bash}
bwa mem
```

We see that `bwa mem` requires a reference genome, input file(s), and has many optional parameters
The reference genome is `chr7.fa`, and the files that start with `chr7.fa.`

We also need to sort the output of `bwa` into the order in which they align in the reference genome.
For this we will use `samtools sort`

```{bash}
samtools sort
```

To avoid writing additional (uncompressed) files, we will combine the alignment and sorting steps together with a unix "pipe".
A pipe `|` takes the output from one program and sends it to the next program.

We can run our `bwa` and `samtools` commands together in a pipe using:

```{bash}
bwa mem chr7.fa normal_R1.fastq.gz normal_R2.fastq.gz | samtools sort - > normal.bam
```

Our variant caller requires a index file for our `normal.bam` file so we create this using `samtools index`:

```{bash}
samtools index normal.bam
```

As we are doing somatic variant calling, we need to repeat these steps on the tumour sequencing reads as well.
Our tumour data is larger than our normal so to reduce the time it takes to perform alignment, we'll tell `bwa` that it can use both CPU cores available to us in our workshop virtual machines: 

```{bash}
bwa mem -t 2 chr7.fa tumour_R1.fastq.gz tumour_R2.fastq.gz | samtools sort - > tumour.bam
samtools index tumour.bam
```

## Variant Calling

We now have both the normal and tumour inputs ready for variant calling.
In a real pipline, there are additional quality control, adapter trimming, and duplicate filtering steps but for this workshop, the data has been pre-processed for us so we can skip these steps.

The variant caller we are using is (Strelka2)[]https://github.com/Illumina/strelka/blob/v2.9.x/docs/userGuide/quickStart.md].
Looking through the quickstart guide, we can see that Strelka2 supports both somatic and germline variant calling.
As we are doing somatic variant calling, we want to configure a somatic Strelka workflow

```{bash}
configureStrelkaSomaticWorkflow.py --normalBam normal.bam --tumorBam tumour.bam --referenceFasta chr7.fa --runDir strelka
```

This created a `strelka` subdirectory containing `runWorkflow.py` which is the command we will use to run Strelka.

```{bash}
strelka/runWorkflow.py
```

As we can see from the usage message, Strelka requires additional arguments.
In our case, we want to specify that we will be running it on the same computer that we are running the `runWorkflow.py` command.
We will also tell Strelka that it can use both the CPU cores available to us.

```{bash}
strelka/runWorkflow.py -j 2 -m local
```

If you navigate to the `colo829/strelka/results/variants/` directory, you should now have a `somatic.snvs.vcf.gz` file containing the somatic variants that Strelka detected in our example sequencing data.