<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genomics Workshop • IntroductionToGenomicsWorkshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genomics Workshop">
<meta property="og:description" content="IntroductionToGenomicsWorkshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IntroductionToGenomicsWorkshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/genomics_workshop.html">Genomics Workshop</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PapenfussLab/IntroductionToGenomicsWorkshop/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="genomics_workshop_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genomics Workshop</h1>
                        <h4 class="author">Ruining Dong<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">James Fu<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
                        <h4 class="author">Daniel Cameron<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PapenfussLab/IntroductionToGenomicsWorkshop/blob/master/vignettes/genomics_workshop.Rmd"><code>vignettes/genomics_workshop.Rmd</code></a></small>
      <div class="hidden name"><code>genomics_workshop.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>In this 2-hour workshop, you will have hands-on experience on SNV calling and analysis. You will be provided toy short-read sequencing samples, followed by read alignments, variant calling and pathogenicity analysis.</p>
<div id="description" class="section level3">
<h3 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h3>
<p>This workshop will guide you through the workflow of a simple genomic SNV analysis.</p>
<p>This is a hands-on workshop, which means you will be given time to get the tools running.</p>
<p>For each task, you will be given a brief instruction prior to getting your hands dirty. At the end of the given time, there will be a brief summary.</p>
</div>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>Basic knowledge of using command line and Rstudio is ideal. Although strictly speaking, no prior knowledge is required to attend this workshop.</p>
</div>
<div id="participation" class="section level3">
<h3 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h3>
<p>The workshop runs for 2 hours. You will find tasks and related information in the worksheet.</p>
<p>If you run into any problems, please describe the issue and send via zoom chat. The instructors will be monitoring the chat panel and answer any question in real time.</p>
</div>
<div id="tools-to-be-used" class="section level3">
<h3 class="hasAnchor">
<a href="#tools-to-be-used" class="anchor"></a>Tools to be used</h3>
<ul>
<li><a href="http://bio-bwa.sourceforge.net/bwa.shtml">bwa</a></li>
<li><a href="http://www.htslib.org/doc/samtools.html">samtools</a></li>
<li><a href="https://github.com/Illumina/strelka">Strelka2</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html">VariantAnnotation</a></li>
<li><a href="https://bioconductor.org/packages/release/data/annotation/html/TxDb.Hsapiens.UCSC.hg19.knownGene.html">TxDb.Hsapiens.UCSC.hg19.knownGene</a></li>
<li><a href="https://bioconductor.org/packages/release/data/annotation/html/BSgenome.Hsapiens.UCSC.hg19.html">BSgenome.Hsapiens.UCSC.hg19</a></li>
<li><a href="https://bioconductor.org/packages/release/data/experiment/html/COSMIC.67.html">COSMIC.67</a></li>
<li><a href="https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html">org.Hs.eg.db</a></li>
<li><a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a></li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<p>For this 120-minute workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Working Environment: command line and Rstudio</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Read Alignment: bwa and samtools</td>
<td>30m</td>
</tr>
<tr class="even">
<td>SNV calling: Strelka2</td>
<td>40m</td>
</tr>
<tr class="odd">
<td>Pathogenicity analysis</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Summary</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<p>Variant calling is a vital component of genomics studies. Genomic variants have been found in various structures and sizes, including single-nucleotide variants (SNVs), small insertion and deletions (INDELs), copy number variants (CNVs) and structural variants (SVs). Computational tools for detecting these variants are developed with different underlying approaches.</p>
<p>This workshop is designed to offer a hands-on experience on genomic variant calling and analysis. The tasks are designed to guide you through the key steps of SNV analysis on a tumour/normal paired sample.</p>
</div>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>To understand the general workflow of a genomic variant analysis</li>
<li>To identify methods to handle short-read sequencing data and variant calls</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>To recall the key concepts in genomic variant analysis</li>
<li>To apply the concepts in SNV calling to germline and paired tumour/normal data</li>
<li>To perform SNV analysis using publicly available resources</li>
</ul>
</div>
</div>
<div id="workshop" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop" class="anchor"></a>Workshop</h2>
<div id="getting-started" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h3>
<ol style="list-style-type: decimal">
<li>Rstudio working environment (console vs. terminal)</li>
<li>Data directory</li>
<li>Working with UNIX commands</li>
</ol>
<p><em>working directories</em></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> colo829</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p><em>running cmd tools</em></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">man</span> samtools <span class="co">#q for quitting the man page</span></span></code></pre></div>
<p><em>piping</em></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="kw">|</span> <span class="fu">head</span> <span class="at">-3</span></span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>SNV analysis workflow</li>
</ol>
<p><img src="../inst/extdata/snvWorkflow.png" width="1156"></p>
</div>
<div id="task-1-read-alignments-sorting-and-indexing" class="section level3">
<h3 class="hasAnchor">
<a href="#task-1-read-alignments-sorting-and-indexing" class="anchor"></a>Task 1: read alignments, sorting, and indexing</h3>
<p>Here you will need to align short-read sequencing reads to the reference genome (hg19) and get them ready for SNV calling. There are two samples you will need the work with, the tumor sample and the matched normal sample of COLO829<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>.</p>
<div id="aligning-reads-to-the-reference-genome" class="section level4">
<h4 class="hasAnchor">
<a href="#aligning-reads-to-the-reference-genome" class="anchor"></a><em>Aligning reads to the reference genome</em>
</h4>
<p>Typical human whole genome sequencing (WGS) data contains hundreds of millions or billions of reads. These data sets take many hours to align so for this workshop we are going to align a very small subset of real sequencing data.</p>
<p>The data we are using today is located in the colo829 directory. The reads can be found at <code>/home/rstudio/colo829/*.fastq.gz</code>, and the reference genome is located at <code>/home/rstudio/colo829/chr7.fa</code>.</p>
<p>Sequence alignment tools are command-line tools so for this first task, we are going to use the command-line terminal.</p>
<p>To start, let us navigate to the directory containing our data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> colo829</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>The <code>cd</code> command is the “change directory” command. It changes our working directory from our “home” directory to the colo829 directory.</p>
<p>The <code>ls</code> command is the “list” command. It outputs all the files and directories in the current directory.</p>
<p>Our input data is paired-end sequencing data for a tumour sample and a matched normal.</p>
<p>This <code>fastq</code> files are output from the sequencing machine and we need to align them to the reference genome. In this workshop, we are going to use chromosome 7 of the hg19 human reference genome as our reference.</p>
<p>We will use <code>bwa mem</code> for our sequence alignment. If we type:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem</span></code></pre></div>
<p>We see that <code>bwa mem</code> requires a reference genome, input file(s), and has many optional parameters. The reference genome is <code>chr7.fa</code>, and the files that start with <code>chr7.fa.</code>.</p>
<details><summary>
Hint
</summary><ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem [options] <span class="op">&lt;</span>idxbase<span class="op">&gt;</span> <span class="op">&lt;</span>in1.fq<span class="op">&gt;</span> [in2.fq] <span class="op">&gt;</span> colo829.sam</span></code></pre></div>
</ul></details>
</div>
<div id="converting-the-output-file-format" class="section level4">
<h4 class="hasAnchor">
<a href="#converting-the-output-file-format" class="anchor"></a><em>Converting the output file format</em>
</h4>
<p>The output file of <code>bwa mem</code> can be saved as a <code>SAM</code> file, standing for Sequence Alignment Map. For storage efficiency, <code>SAM</code> files are often losslessly compressed to <code>BAM</code> format, standing for Binary Alignment Map.</p>
<p>Try it yourself by modifying this template:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-S</span> <span class="at">-b</span> output.sam <span class="at">-o</span> output.bam</span></code></pre></div>
<p>This step is optional if you “pipe” your <code>bwa</code> output to <code>samtools</code> (see below).</p>
</div>
<div id="sorting-and-indexing" class="section level4">
<h4 class="hasAnchor">
<a href="#sorting-and-indexing" class="anchor"></a><em>Sorting and indexing</em>
</h4>
<p>We also need to sort the output of <code>bwa</code> into the order in which they align in the reference genome. For this we will use <code>samtools sort</code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort</span></code></pre></div>
<p>and <code>samtools index</code>:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index</span></code></pre></div>
<p>To avoid writing additional (uncompressed) files, we will combine the alignment and sorting steps together with a unix “pipe”. A pipe <code><a href="https://rdrr.io/r/base/Logic.html">|</a></code> takes the output from one program and sends it to the next program.</p>
<p>We can run our <code>bwa</code> and <code>samtools</code> commands together in a pipe using:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem chr7.fa normal_R1.fastq.gz normal_R2.fastq.gz <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">-</span> <span class="op">&gt;</span> normal.bam</span></code></pre></div>
<p>Our variant caller requires a index file for our <code>normal.bam</code> file so we create this using <code>samtools index</code>:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index normal.bam</span></code></pre></div>
<p>As we are doing somatic variant calling, we need to repeat these steps on the tumour sequencing reads as well. Our tumour data is larger than our normal so to reduce the time it takes to perform alignment, we’ll tell <code>bwa mem</code> that it can use both CPU cores available to us in our workshop virtual machines by telling it to use 2 threads.</p>
<p>Now please try applying all the above steps on the tumour reads.</p>
<details><summary>
Solution
</summary><ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem <span class="at">-t</span> 2 chr7.fa tumour_R1.fastq.gz tumour_R2.fastq.gz <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">-</span> <span class="op">&gt;</span> tumour.bam</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index tumour.bam</span></code></pre></div>
</ul></details>
</div>
</div>
<div id="task-2-calling-snvs-using-strelka2" class="section level3">
<h3 class="hasAnchor">
<a href="#task-2-calling-snvs-using-strelka2" class="anchor"></a>Task 2: calling SNVs using Strelka2</h3>
<p>Now that we have the sequencing reads in order, we will start calling SNVs. We now have both the normal and tumour inputs ready for variant calling. In a real pipeline, there are additional quality control, adapter trimming, and duplicate filtering steps but for this workshop, the data has been pre-processed for us so we can skip these steps.</p>
<p>The variant caller we are using is <a href="https://github.com/Illumina/strelka/blob/v2.9.x/docs/userGuide/quickStart.md">Strelka2</a>. Looking through the quick start guide, we can see that Strelka2 supports both somatic and germline variant calling. As we are doing somatic variant calling, we want to configure a somatic Strelka workflow</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">configureStrelkaSomaticWorkflow.py</span> <span class="dt">\</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--normalBam</span> normal.bam <span class="dt">\</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--tumorBam</span> tumour.bam <span class="dt">\</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--referenceFasta</span> chr7.fa <span class="dt">\</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--runDir</span> strelka</span></code></pre></div>
<p>This created a <code>strelka</code> subdirectory containing <code>runWorkflow.py</code> which is the command we will use to run Strelka.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">strelka/runWorkflow.py</span></span></code></pre></div>
<p>As we can see from the usage message, Strelka requires additional arguments. In our case, we want to specify that we will be running it on the same computer that we are running the <code>runWorkflow.py</code> command. We will also tell Strelka that it can use both the CPU cores available to us.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">strelka/runWorkflow.py</span> <span class="at">-j</span> 2 <span class="at">-m</span> local</span></code></pre></div>
<p>If you navigate to the <code>colo829/strelka/results/variants/</code> directory, you should now have a <code>somatic.snvs.vcf.gz</code> file containing the somatic variants that Strelka detected in our example sequencing data.</p>
</div>
<div id="task-3-investigating-the-driver-gene" class="section level3">
<h3 class="hasAnchor">
<a href="#task-3-investigating-the-driver-gene" class="anchor"></a>Task 3: investigating the driver gene</h3>
<div id="load-packages" class="section level4">
<h4 class="hasAnchor">
<a href="#load-packages" class="anchor"></a><em>Load packages</em>
</h4>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">VariantAnnotation</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></code></pre></div>
</div>
<div id="read-and-pre-process-vcf-file" class="section level4">
<h4 class="hasAnchor">
<a href="#read-and-pre-process-vcf-file" class="anchor"></a><em>Read and pre-process VCF file</em>
</h4>
<p>We use the <code>VariantAnnotation</code> package to read and annotate the SNVs we just found.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#vcf &lt;- readVcf(system.file("extdata", "strelka.somatic.vcf", package = #"IntroductionToGenomicsWorkshop"))</span>
<span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html">readVcf</a></span><span class="op">(</span><span class="st">"../inst/extdata/strelka.somatic.vcf"</span><span class="op">)</span></code></pre></div>
<p>Variant sequence style is changed to maintain its consistency with other databases.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></code></pre></div>
</div>
<div id="header-information-of-vcf-file" class="section level4">
<h4 class="hasAnchor">
<a href="#header-information-of-vcf-file" class="anchor"></a><em>Header information of VCF file</em>
</h4>
<p><code><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html">header()</a></code> function will show the header information from the VCF file.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html">header</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>
<span class="co">#&gt; class: VCFHeader </span>
<span class="co">#&gt; samples(2): NORMAL TUMOR</span>
<span class="co">#&gt; meta(12): cmdline content ... startTime contig</span>
<span class="co">#&gt; fixed(1): FILTER</span>
<span class="co">#&gt; info(22): AC AF ... TQSS_NT set</span>
<span class="co">#&gt; geno(15): AU CU ... TOR TU</span></code></pre></div>
<p>Further information for the headers can be extracted by functions like <code>samples(), geno(), meta()</code>, etc.</p>
<details><summary>
Solution
</summary><ul>
<li>
This will fail.
</li>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html">samples</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>
<span class="co">#&gt; Error in (function (classes, fdef, mtable) : unable to find an inherited method for function 'samples' for signature '"CollapsedVCF"'</span></code></pre></div>
<li>
These will work.
</li>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html">samples</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html">header</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "NORMAL" "TUMOR"</span>
<span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">geno</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>
<span class="co">#&gt; List of length 15</span>
<span class="co">#&gt; names(15): AU CU DP DP2 DP50 FDP FDP50 GU SDP SUBDP SUBDP50 TAR TIR TOR TU</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">geno</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCFHeader-class.html">header</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 3 rows and 3 columns</span>
<span class="co">#&gt;         Number        Type            Description</span>
<span class="co">#&gt;    &lt;character&gt; &lt;character&gt;            &lt;character&gt;</span>
<span class="co">#&gt; AU           2     Integer Number of 'A' allele..</span>
<span class="co">#&gt; CU           2     Integer Number of 'C' allele..</span>
<span class="co">#&gt; DP           1     Integer   Read depth for tier1</span></code></pre></div>
</ul></details>
</div>
<div id="variant-genomic-position" class="section level4">
<h4 class="hasAnchor">
<a href="#variant-genomic-position" class="anchor"></a><em>Variant genomic position</em>
</h4>
<p>The chromosomal location for each SNV can be identified by <code>rowRanges()</code>.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 5 metadata columns:</span>
<span class="co">#&gt;                seqnames    ranges strand | paramRangeID            REF</span>
<span class="co">#&gt;                   &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;factor&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:10250_A/C     chr1     10250      * |           NA              A</span>
<span class="co">#&gt;    1:10257_A/C     chr1     10257      * |           NA              A</span>
<span class="co">#&gt;   1:533197_A/G     chr1    533197      * |           NA              A</span>
<span class="co">#&gt;                               ALT      QUAL      FILTER</span>
<span class="co">#&gt;                &lt;DNAStringSetList&gt; &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt;    1:10250_A/C                  C        NA        PASS</span>
<span class="co">#&gt;    1:10257_A/C                  C        NA        PASS</span>
<span class="co">#&gt;   1:533197_A/G                  G        NA        PASS</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome</span></code></pre></div>
<p>Individual fields can be pulled out either with the <code><a href="https://rdrr.io/r/base/Extract.html">$</a></code> sign or named accessors like <code><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">ref()</a></code>, <code><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">alt()</a></code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">$</span><span class="va">ALT</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; DNAStringSetList of length 3</span>
<span class="co">#&gt; [[1]] C</span>
<span class="co">#&gt; [[2]] C</span>
<span class="co">#&gt; [[3]] G</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">alt</a></span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; DNAStringSetList of length 3</span>
<span class="co">#&gt; [[1]] C</span>
<span class="co">#&gt; [[2]] C</span>
<span class="co">#&gt; [[3]] G</span></code></pre></div>
</div>
<div id="locate-variants-in-and-around-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#locate-variants-in-and-around-genes" class="anchor"></a><em>Locate variants in and around genes</em>
</h4>
<p>Variant location with respect to genes can be identified with the <code><a href="https://rdrr.io/pkg/VariantAnnotation/man/locateVariants-methods.html">locateVariants()</a></code> function.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get variant location information.</span>
<span class="va">rd</span> <span class="op">&lt;-</span> <span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>
<span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span>
<span class="va">loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/locateVariants-methods.html">locateVariants</a></span><span class="op">(</span><span class="va">rd</span>, <span class="va">txdb</span>, <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VariantType-class.html">CodingVariants</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">loc</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 9 metadata columns:</span>
<span class="co">#&gt;                seqnames    ranges strand | LOCATION  LOCSTART    LOCEND</span>
<span class="co">#&gt;                   &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   1:883516_G/A     chr1    883516      - |   coding      1045      1045</span>
<span class="co">#&gt;   1:883516_G/A     chr1    883516      - |   coding      1654      1654</span>
<span class="co">#&gt;   1:883516_G/A     chr1    883516      - |   coding      1654      1654</span>
<span class="co">#&gt;                  QUERYID        TXID         CDSID      GENEID       PRECEDEID</span>
<span class="co">#&gt;                &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;</span>
<span class="co">#&gt;   1:883516_G/A        18        4114         12095       26155                </span>
<span class="co">#&gt;   1:883516_G/A        18        4115         12095       26155                </span>
<span class="co">#&gt;   1:883516_G/A        18        4116         12095       26155                </span>
<span class="co">#&gt;                       FOLLOWID</span>
<span class="co">#&gt;                &lt;CharacterList&gt;</span>
<span class="co">#&gt;   1:883516_G/A                </span>
<span class="co">#&gt;   1:883516_G/A                </span>
<span class="co">#&gt;   1:883516_G/A                </span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Gene ID can be replaced my gene symbols which are more readable, using <code>org.HS.eg.db</code> database.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_symbol</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/select.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys<span class="op">=</span><span class="va">loc</span><span class="op">$</span><span class="va">GENEID</span>, columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span><span class="op">)</span>, keytype<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">gene_symbol</span>, <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; [1] "NOC2L"  "NOC2L"  "NOC2L"  "PRDM16" "PRDM16"</span>
<span class="va">loc</span><span class="op">$</span><span class="va">SYMBOl</span> <span class="op">&lt;-</span> <span class="va">gene_symbol</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">loc</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 10 metadata columns:</span>
<span class="co">#&gt;                seqnames    ranges strand | LOCATION  LOCSTART    LOCEND</span>
<span class="co">#&gt;                   &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   1:883516_G/A     chr1    883516      - |   coding      1045      1045</span>
<span class="co">#&gt;   1:883516_G/A     chr1    883516      - |   coding      1654      1654</span>
<span class="co">#&gt;   1:883516_G/A     chr1    883516      - |   coding      1654      1654</span>
<span class="co">#&gt;                  QUERYID        TXID         CDSID      GENEID       PRECEDEID</span>
<span class="co">#&gt;                &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;</span>
<span class="co">#&gt;   1:883516_G/A        18        4114         12095       26155                </span>
<span class="co">#&gt;   1:883516_G/A        18        4115         12095       26155                </span>
<span class="co">#&gt;   1:883516_G/A        18        4116         12095       26155                </span>
<span class="co">#&gt;                       FOLLOWID      SYMBOl</span>
<span class="co">#&gt;                &lt;CharacterList&gt; &lt;character&gt;</span>
<span class="co">#&gt;   1:883516_G/A                       NOC2L</span>
<span class="co">#&gt;   1:883516_G/A                       NOC2L</span>
<span class="co">#&gt;   1:883516_G/A                       NOC2L</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The number of SNVs on each chromosome can be displayed by <code>seqnames()</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqnames</span><span class="op">(</span><span class="va">loc</span><span class="op">)</span>
<span class="co">#&gt; factor-Rle of length 918 with 23 runs</span>
<span class="co">#&gt;   Lengths:    94    99    53    55    31 ...    66    12    10    14    27</span>
<span class="co">#&gt;   Values : chr1  chr2  chr3  chr4  chr5  ... chr19 chr20 chr21 chr22 chrX </span>
<span class="co">#&gt; Levels(25): chr1 chr2 chr3 chr4 chr5 chr6 ... chr20 chr21 chr22 chrX chrY chrM</span></code></pre></div>
<p>The documentation of <code><a href="https://rdrr.io/pkg/VariantAnnotation/man/locateVariants-methods.html">locateVariants()</a></code> describes how to identify variants in other regions, like <em>splicing sites</em>.</p>
<details><summary>
Solution
</summary><ul>
<li>
Check the documentation of <code><a href="https://rdrr.io/pkg/VariantAnnotation/man/locateVariants-methods.html">locateVariants()</a></code>.
</li>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">?</span><span class="va">locateVariants</span></code></pre></div>
<li>
<p>Following functions are used to identify variants in distinct gene regions:</p>
<table class="table">
<colgroup>
<col width="23%">
<col width="76%">
</colgroup>
<thead><tr class="header">
<th>Function</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>CodingVariants()</td>
<td>Falls within a coding region</td>
</tr>
<tr class="even">
<td>IntronVariants()</td>
<td>Falls within an intron region</td>
</tr>
<tr class="odd">
<td>FiveUTRVariants()</td>
<td>Falls within a 5’ untranslated region</td>
</tr>
<tr class="even">
<td>ThreeUTRVariants()</td>
<td>Falls within a 3’ untranslated region</td>
</tr>
<tr class="odd">
<td>IntergenicVariants()</td>
<td>Does not fall within a transcript associated with a gene</td>
</tr>
<tr class="even">
<td>SpliceSiteVariants()</td>
<td>Overlaps any portion of the first 2 or last 2 nucleotide of an intron</td>
</tr>
<tr class="odd">
<td>PromoterVariants()</td>
<td>Falls within a promoter region of a transcript</td>
</tr>
<tr class="even">
<td>AllVariants()</td>
<td>All regions</td>
</tr>
</tbody>
</table>
</li>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">loc_splice</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/locateVariants-methods.html">locateVariants</a></span><span class="op">(</span><span class="va">rd</span>, <span class="va">txdb</span>, <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VariantType-class.html">SpliceSiteVariants</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">loc_splice</span>, <span class="fl">3</span><span class="op">)</span>
<span class="co">#&gt; GRanges object with 3 ranges and 9 metadata columns:</span>
<span class="co">#&gt;                seqnames    ranges strand |   LOCATION  LOCSTART    LOCEND</span>
<span class="co">#&gt;                   &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |   &lt;factor&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   1:787492_T/C     chr1    787492      + | spliceSite      &lt;NA&gt;      &lt;NA&gt;</span>
<span class="co">#&gt;   1:787492_T/C     chr1    787492      + | spliceSite      &lt;NA&gt;      &lt;NA&gt;</span>
<span class="co">#&gt;   1:787492_T/C     chr1    787492      + | spliceSite      &lt;NA&gt;      &lt;NA&gt;</span>
<span class="co">#&gt;                  QUERYID      TXID         CDSID      GENEID       PRECEDEID</span>
<span class="co">#&gt;                &lt;integer&gt; &lt;integer&gt; &lt;IntegerList&gt; &lt;character&gt; &lt;CharacterList&gt;</span>
<span class="co">#&gt;   1:787492_T/C        12        15                    643837                </span>
<span class="co">#&gt;   1:787492_T/C        12        16                    643837                </span>
<span class="co">#&gt;   1:787492_T/C        12        18                    643837                </span>
<span class="co">#&gt;                       FOLLOWID</span>
<span class="co">#&gt;                &lt;CharacterList&gt;</span>
<span class="co">#&gt;   1:787492_T/C                </span>
<span class="co">#&gt;   1:787492_T/C                </span>
<span class="co">#&gt;   1:787492_T/C                </span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span>
<span class="fu">seqnames</span><span class="op">(</span><span class="va">loc_splice</span><span class="op">)</span>
<span class="co">#&gt; factor-Rle of length 38 with 7 runs</span>
<span class="co">#&gt;   Lengths:     6     7    13     1     4     1     6</span>
<span class="co">#&gt;   Values : chr1  chr2  chr3  chr4  chr9  chr11 chr16</span>
<span class="co">#&gt; Levels(25): chr1 chr2 chr3 chr4 chr5 chr6 ... chr20 chr21 chr22 chrX chrY chrM</span></code></pre></div>
</ul></details>
</div>
<div id="amino-acid-coding-changes" class="section level4">
<h4 class="hasAnchor">
<a href="#amino-acid-coding-changes" class="anchor"></a><em>Amino acid coding changes</em>
</h4>
<p><code>predictCoding</code> computes amino acid coding changes for non-synonymous mutations.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coding</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/predictCoding-methods.html">predictCoding</a></span><span class="op">(</span><span class="va">vcf</span>, <span class="va">txdb</span>, seqSource <span class="op">=</span> <span class="va">Hsapiens</span><span class="op">)</span>
<span class="va">coding</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="co">#&gt; GRanges object with 2 ranges and 17 metadata columns:</span>
<span class="co">#&gt;                 seqnames    ranges strand | paramRangeID            REF</span>
<span class="co">#&gt;                    &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;factor&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A     chr1    883516      - |           NA              G</span>
<span class="co">#&gt;   1:3329312_C/T     chr1   3329312      + |           NA              C</span>
<span class="co">#&gt;                                ALT      QUAL      FILTER      varAllele</span>
<span class="co">#&gt;                 &lt;DNAStringSetList&gt; &lt;numeric&gt; &lt;character&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A                  A        NA        PASS              T</span>
<span class="co">#&gt;   1:3329312_C/T                  T        NA        PASS              T</span>
<span class="co">#&gt;                    CDSLOC    PROTEINLOC   QUERYID        TXID         CDSID</span>
<span class="co">#&gt;                 &lt;IRanges&gt; &lt;IntegerList&gt; &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt;</span>
<span class="co">#&gt;    1:883516_G/A      1654           552        18        4116         12095</span>
<span class="co">#&gt;   1:3329312_C/T      2551           851        79         154           395</span>
<span class="co">#&gt;                      GENEID   CONSEQUENCE       REFCODON       VARCODON</span>
<span class="co">#&gt;                 &lt;character&gt;      &lt;factor&gt; &lt;DNAStringSet&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A       26155 synonymous               CTG            TTG</span>
<span class="co">#&gt;   1:3329312_C/T       63976 nonsynonymous            CCC            TCC</span>
<span class="co">#&gt;                         REFAA         VARAA</span>
<span class="co">#&gt;                 &lt;AAStringSet&gt; &lt;AAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A             L             L</span>
<span class="co">#&gt;   1:3329312_C/T             P             S</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome</span></code></pre></div>
<p>Here shows all consequences of coding variants.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">coding</span><span class="op">$</span><span class="va">CONSEQUENCE</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    frameshift      nonsense nonsynonymous    synonymous </span>
<span class="co">#&gt;            21            46           562           289</span></code></pre></div>
<p>We can also annotate each of these variants with their corresponding gene by looking up the <code>GENEID</code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coding</span><span class="op">$</span><span class="va">SYMBOL</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/select.html">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>, keys<span class="op">=</span><span class="va">coding</span><span class="op">$</span><span class="va">GENEID</span>, columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span><span class="op">)</span>, keytype<span class="op">=</span><span class="st">"ENTREZID"</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span>
<span class="co">#&gt; 'select()' returned many:1 mapping between keys and columns</span>
<span class="va">coding</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="co">#&gt; GRanges object with 2 ranges and 18 metadata columns:</span>
<span class="co">#&gt;                 seqnames    ranges strand | paramRangeID            REF</span>
<span class="co">#&gt;                    &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;factor&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A     chr1    883516      - |           NA              G</span>
<span class="co">#&gt;   1:3329312_C/T     chr1   3329312      + |           NA              C</span>
<span class="co">#&gt;                                ALT      QUAL      FILTER      varAllele</span>
<span class="co">#&gt;                 &lt;DNAStringSetList&gt; &lt;numeric&gt; &lt;character&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A                  A        NA        PASS              T</span>
<span class="co">#&gt;   1:3329312_C/T                  T        NA        PASS              T</span>
<span class="co">#&gt;                    CDSLOC    PROTEINLOC   QUERYID        TXID         CDSID</span>
<span class="co">#&gt;                 &lt;IRanges&gt; &lt;IntegerList&gt; &lt;integer&gt; &lt;character&gt; &lt;IntegerList&gt;</span>
<span class="co">#&gt;    1:883516_G/A      1654           552        18        4116         12095</span>
<span class="co">#&gt;   1:3329312_C/T      2551           851        79         154           395</span>
<span class="co">#&gt;                      GENEID   CONSEQUENCE       REFCODON       VARCODON</span>
<span class="co">#&gt;                 &lt;character&gt;      &lt;factor&gt; &lt;DNAStringSet&gt; &lt;DNAStringSet&gt;</span>
<span class="co">#&gt;    1:883516_G/A       26155 synonymous               CTG            TTG</span>
<span class="co">#&gt;   1:3329312_C/T       63976 nonsynonymous            CCC            TCC</span>
<span class="co">#&gt;                         REFAA         VARAA      SYMBOL</span>
<span class="co">#&gt;                 &lt;AAStringSet&gt; &lt;AAStringSet&gt; &lt;character&gt;</span>
<span class="co">#&gt;    1:883516_G/A             L             L       NOC2L</span>
<span class="co">#&gt;   1:3329312_C/T             P             S      PRDM16</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences from an unspecified genome</span></code></pre></div>
<p>We can discard the synonymous mutations, and the rest are potential cancer drivers. Then map them back to the VCF data and filter it.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">potential_drivers</span> <span class="op">=</span> <span class="va">coding</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">coding</span><span class="op">$</span><span class="va">CONSEQUENCE</span><span class="op">)</span> <span class="op">&amp;</span> <span class="va">coding</span><span class="op">$</span><span class="va">CONSEQUENCE</span> <span class="op">!=</span> <span class="st">"synonymous"</span><span class="op">]</span>
<span class="va">potential_drivers</span> <span class="op">=</span> <span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span><span class="op">[</span><span class="fu">overlapsAny</span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">vcf</span><span class="op">)</span>, <span class="va">potential_drivers</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="predict-impact" class="section level4">
<h4 class="hasAnchor">
<a href="#predict-impact" class="anchor"></a><em>Predict impact</em>
</h4>
<p>For the nonsynonymous mutations, we’d like to know how much these impact the protein. For this, we will use to pre-calculated scores from the rfPred package. To work out how to use the package, we look up the help.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">`?rfPred-package`</span></code></pre></div>
<p>If you click on the index link, it will show you all functions available in the package. Check what the scoring function requires we see the <code>variant_list</code> parameter requires a <code>data.frame</code> with specific columns. We’ll need to create that before calling the function.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.sbim.fr/rfPred">rfPred</a></span><span class="op">)</span>
<span class="va">rfpred_gr</span> <span class="op">=</span> <span class="va">potential_drivers</span>
<span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">rfpred_gr</span><span class="op">)</span> <span class="op">=</span> <span class="st">"NCBI"</span>
<span class="va">rfpred_df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  seqnames<span class="op">=</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">rfpred_gr</span><span class="op">)</span>,
  start<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">rfpred_gr</span><span class="op">)</span>,
  reference<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">rfpred_gr</span><span class="op">$</span><span class="va">REF</span><span class="op">)</span>,
  alteration <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">rfpred_gr</span><span class="op">$</span><span class="va">ALT</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">scores</span> <span class="op">=</span> <span class="fu">rfPred</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rfPred/man/rfPred_scores-methods.html">rfPred_scores</a></span><span class="op">(</span>
  variant_list<span class="op">=</span><span class="va">rfpred_df</span>, 
  data<span class="op">=</span><span class="st">"/home/rstudio/rfPred/all_chr_rfPred.txtz"</span>,
  index<span class="op">=</span><span class="st">"/home/rstudio/rfPred/all_chr_rfPred.txtz.tbi"</span><span class="op">)</span>
<span class="va">scores</span> <span class="op">=</span> <span class="va">scores</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">scores</span><span class="op">$</span><span class="va">rfPred_score</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">scores</span>, <span class="fl">20</span><span class="op">)</span></code></pre></div>
</div>
<div id="match-variants-within-cosmic" class="section level4">
<h4 class="hasAnchor">
<a href="#match-variants-within-cosmic" class="anchor"></a><em>Match variants within COSMIC</em>
</h4>
<p>To work out which variants are likely driver genes, we can check if they have been seen before in <a href="https://cancer.sanger.ac.uk/cosmic/">COSMIC</a></p>
<p>Read and pre-process COSMIC database.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cosmic_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"vcf"</span>, <span class="st">"cosmic_67.vcf.gz"</span>, package <span class="op">=</span> <span class="st">"COSMIC.67"</span><span class="op">)</span>
<span class="va">cosmic_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/readVcf-methods.html">readVcf</a></span><span class="op">(</span><span class="va">cosmic_path</span>, genome <span class="op">=</span> <span class="st">"GRCh37"</span><span class="op">)</span>
<span class="fu">seqlevelsStyle</span><span class="op">(</span><span class="va">cosmic_db</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"UCSC"</span></code></pre></div>
<p>The potential driver variants are matched with the COSMIC database.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Match by chromosome and location of the variants.</span>
<span class="va">cosmics_hits</span> <span class="op">=</span> <span class="fu">findOverlaps</span><span class="op">(</span><span class="va">potential_drivers</span>, <span class="fu">rowRanges</span><span class="op">(</span><span class="va">cosmic_db</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"equal"</span><span class="op">)</span>
<span class="co">#&gt; Warning in .Seqinfo.mergexy(x, y): Each of the 2 combined objects has sequence levels not in the other:</span>
<span class="co">#&gt;   - in 'x': chrM</span>
<span class="co">#&gt;   - in 'y': chrMT</span>
<span class="co">#&gt;   Make sure to always combine/compare objects based on the same reference</span>
<span class="co">#&gt;   genome (use suppressWarnings() to suppress this warning).</span>
<span class="co"># Match by reference and alternative allele of the variants.</span>
<span class="va">cosmics_hits</span> <span class="op">=</span> <span class="va">cosmics_hits</span><span class="op">[</span>
  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">potential_drivers</span><span class="op">$</span><span class="va">REF</span><span class="op">[</span><span class="fu">queryHits</span><span class="op">(</span><span class="va">cosmics_hits</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">cosmic_db</span><span class="op">)</span><span class="op">$</span><span class="va">REF</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">cosmics_hits</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">&amp;</span>
  <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">potential_drivers</span><span class="op">$</span><span class="va">ALT</span><span class="op">[</span><span class="fu">queryHits</span><span class="op">(</span><span class="va">cosmics_hits</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="fu">rowRanges</span><span class="op">(</span><span class="va">cosmic_db</span><span class="op">)</span><span class="op">$</span><span class="va">ALT</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">cosmics_hits</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  <span class="op">]</span></code></pre></div>
<p>Finally, we take the COSMIC information of the potential driver mutants and rank them by <code>CNT</code>. <code>CNT</code> values denote the sample count for the given variant.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">found_in_cosmic</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/VariantAnnotation/man/VCF-class.html">info</a></span><span class="op">(</span><span class="va">cosmic_db</span><span class="op">)</span><span class="op">[</span><span class="fu">subjectHits</span><span class="op">(</span><span class="va">cosmics_hits</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">found_in_cosmic</span> <span class="op">=</span> <span class="va">found_in_cosmic</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">found_in_cosmic</span><span class="op">$</span><span class="va">CNT</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">found_in_cosmic</span> <span class="op">=</span> <span class="va">found_in_cosmic</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">found_in_cosmic</span><span class="op">$</span><span class="va">CNT</span><span class="op">)</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">found_in_cosmic</span>, <span class="fl">5</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 5 rows and 5 columns</span>
<span class="co">#&gt;                              GENE      STRAND         CDS          AA       CNT</span>
<span class="co">#&gt;                       &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;integer&gt;</span>
<span class="co">#&gt; COSM476                      BRAF           -   c.1799T&gt;A     p.V600E     19691</span>
<span class="co">#&gt; COSM25632                   AGFG1           +   c.1018G&gt;A     p.G340R         7</span>
<span class="co">#&gt; COSM1405984 AGFG1_ENST00000409979           +   c.1090G&gt;A     p.G364R         5</span>
<span class="co">#&gt; COSM226579  STK19_ENST00000375333           +    c.265G&gt;A      p.D89N         5</span>
<span class="co">#&gt; COSM23305                   HDAC2           -   c.1225C&gt;T     p.R409*         4</span></code></pre></div>
<p>Now you can search the mutation identifier <em>(COSM476 for example)</em> in <a href="https://cancer.sanger.ac.uk/cosmic/mutation/overview?id=170110762#">COSMIC</a> and learn more about the variant.</p>
<!-- <details> -->
<!-- <summary>Solution </summary> -->
<!-- <ul> -->
<!-- <li>First do this.</li> -->
<!-- ```{bash, eval = FALSE} -->
<!-- #bwa index ref.fa -->
<!-- bwa mem ref.fa read1.fq read2.fq > output.sam -->
<!-- ``` -->
<!-- <li>Then do that.</li> -->
<!-- ```{bash, eval = FALSE} -->
<!-- samtools view -S -b output.sam -o output.bam #converting SAM to BAM -->
<!-- samtools sort output.bam output.sorted #sorting the BAM reads -->
<!-- samtools index output.sorted.bam #indexing the sorted BAM -->
<!-- ``` -->
<!-- <li>OR, pipe some of the commands in a one-liner:</li> -->
<!-- ```{bash, eval = FALSE} -->
<!-- bwa mem ref.fa read1.fq read2.fq | samtools view -S -b - | samtools sort -o output.sorted.bam - -->
<!-- samtools index output.sorted.bam -->
<!-- ``` -->
<!-- <li>Blah blah.</li> -->
<!-- <li>You should be able to see the resultant files `file1` and `file2`.</li> -->
<!-- </ul> -->
<!-- </details> -->
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:dong.rn@wehi.edu.au" class="email">dong.rn@wehi.edu.au</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:fu.j@wehi.edu.au" class="email">fu.j@wehi.edu.au</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="mailto:cameron.d@wehi.edu.au" class="email">cameron.d@wehi.edu.au</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>Craig, D., Nasser, S., Corbett, R. et al. A somatic reference standard for cancer genome sequencing. Sci Rep 6, 24607 (2016). <a href="https://doi.org/10.1038/srep24607" class="uri">https://doi.org/10.1038/srep24607</a><a href="#fnref4" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Cameron, Ruining Dong, James Fu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
