<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Genomics Workshop • IntroductionToGenomicsWorkshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Genomics Workshop">
<meta property="og:description" content="IntroductionToGenomicsWorkshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IntroductionToGenomicsWorkshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Genomics_workshop_task1.html">Genomics Workshop: Task 1</a>
    </li>
    <li>
      <a href="../articles/Genomics_workshop_task2.html">Genomics Workshop: Task 2</a>
    </li>
    <li>
      <a href="../articles/Genomics_workshop_task3.html">Genomics Workshop: Task 3</a>
    </li>
    <li>
      <a href="../articles/genomics_workshop.html">Genomics Workshop</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/PapenfussLab/IntroductionToGenomicsWorkshop/">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="genomics_workshop_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Genomics Workshop</h1>
                        <h4 class="author">Ruining Dong<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
                        <h4 class="author">James Fu<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
                        <h4 class="author">Daniel Cameron<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/PapenfussLab/IntroductionToGenomicsWorkshop/blob/master/vignettes/genomics_workshop.Rmd"><code>vignettes/genomics_workshop.Rmd</code></a></small>
      <div class="hidden name"><code>genomics_workshop.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>In this 1-hour workshop, you will have hands-on experience on SNV calling and analysis. You will be provided toy short-read sequencing samples, followed by read alignments, variant calling and pathogenicity analysis.</p>
<div id="description" class="section level3">
<h3 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h3>
<p>This workshop will guide you through the workflow of a simple genomic SNV analysis.</p>
<p>This is a hands-on workshop, which means you will be given time to get the tools running.</p>
<p>For each task, you will be given a brief instruction prior to getting your hands dirty. At the end of the given time, there will be a brief summary.</p>
</div>
<div id="pre-requisites" class="section level3">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>Basic knowledge of using command line and Rstudio is ideal. Although strictly speaking, no prior knowledge is required to attend this workshop.</p>
</div>
<div id="participation" class="section level3">
<h3 class="hasAnchor">
<a href="#participation" class="anchor"></a>Participation</h3>
<p>The workshop runs for 2 hours. You will find tasks and related information in the worksheet.</p>
<p>If you run into any problems, please describe the issue and send via zoom chat. The instructors will be monitoring the chat panel and answer any question in real time.</p>
</div>
<div id="tools-to-be-used" class="section level3">
<h3 class="hasAnchor">
<a href="#tools-to-be-used" class="anchor"></a>Tools to be used</h3>
<ul>
<li><a href="http://bio-bwa.sourceforge.net/bwa.shtml">bwa</a></li>
<li><a href="http://www.htslib.org/doc/samtools.html">samtools</a></li>
<li><a href="https://github.com/Illumina/strelka">Strelka2</a></li>
<li>
<a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a> <strong>To be updated</strong>
</li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h3>
<p>For this 120-minute workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Working Environment: command line and Rstudio</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Read Alignment: bwa and samtools</td>
<td>30m</td>
</tr>
<tr class="even">
<td>SNV calling: Strelka2</td>
<td>40m</td>
</tr>
<tr class="odd">
<td>Pathogenicity analysis</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Summary</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h3>
<p>Variant calling is a vital component of genomics studies. Genomic variants have been found in various structures and sizes, including single-nucleotide variants (SNVs), small insertion and deletions (INDELs), copy number variants (CNVs) and structural variants (SVs). Computational tools for detecting these variants are developed with different underlying approaches.</p>
<p>This workshop is designed to offer a hands-on experience on genomic variant calling and analysis. The tasks are designed to guide you through the key steps of SNV analysis on a tumour/normal paired sample.</p>
</div>
<div id="learning-goals" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>To understand the general workflow of a genomic variant analysis</li>
<li>To identify methods to handle short-read sequencing data and variant calls</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>To recall the key concepts in genomic variant analysis</li>
<li>To apply the concepts in SNV calling to germline and paired tumour/normal data</li>
<li>To perform SNV analysis using publicly available resources</li>
</ul>
</div>
</div>
<div id="workshop" class="section level2">
<h2 class="hasAnchor">
<a href="#workshop" class="anchor"></a>Workshop</h2>
<div id="getting-started" class="section level3">
<h3 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started:</h3>
<ol style="list-style-type: decimal">
<li>Rstudio working environment (console vs. terminal)</li>
<li>Data directory</li>
<li>Working with UNIX commands</li>
<li>SNV analysis workflow (<strong>image to be updated</strong>)</li>
</ol>
<p><img src="../inst/images/Example-of-a-best-practices-SNV-calling-workflow-for-somatic-exome-and-genome-data.png" width="425"></p>
</div>
<div id="task-1-read-alignments-sorting-and-indexing" class="section level3">
<h3 class="hasAnchor">
<a href="#task-1-read-alignments-sorting-and-indexing" class="anchor"></a>Task 1: read alignments, sorting, and indexing</h3>
<p>Here you will need to align short-read sequencing reads to the reference genome (hg19) and get them ready for SNV calling. There are two samples you will need the work with, the tumor sample and the matched normal sample of COLO829<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>.</p>
<div id="aligning-reads-to-the-reference-genome" class="section level4">
<h4 class="hasAnchor">
<a href="#aligning-reads-to-the-reference-genome" class="anchor"></a><em>Aligning reads to the reference genome</em>
</h4>
<p>First, we need to map the sequencing reads to the reference genome. The reads can be found at <code>/path/to/fastq</code>, and the reference genome is located at <code>/path/to/fa</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem [options] <span class="op">&lt;</span>idxbase<span class="op">&gt;</span> <span class="op">&lt;</span>in1.fq<span class="op">&gt;</span> [in2.fq] <span class="op">&gt;</span> colo829.sam</span></code></pre></div>
</div>
<div id="converting-the-output-file-format" class="section level4">
<h4 class="hasAnchor">
<a href="#converting-the-output-file-format" class="anchor"></a><em>Converting the output file format</em>
</h4>
<p>To be updated</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#bwa mem [options] &lt;idxbase&gt; &lt;in1.fq&gt; [in2.fq] &gt; colo829.sam</span></span></code></pre></div>
</div>
<div id="sorting-and-indexing" class="section level4">
<h4 class="hasAnchor">
<a href="#sorting-and-indexing" class="anchor"></a><em>Sorting and indexing</em>
</h4>
<p>To be updated</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#bwa mem [options] &lt;idxbase&gt; &lt;in1.fq&gt; [in2.fq] &gt; colo829.sam</span></span></code></pre></div>
<details><summary>
Solution (interactively)
</summary><ul>
<li>
First do this.
</li>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> index ref.fa</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem ref.fa read1.fq read2.fq <span class="op">&gt;</span> output.sam</span></code></pre></div>
<li>
Then do that.
</li>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-S</span> <span class="at">-b</span> output.sam <span class="at">-o</span> output.bam <span class="co">#converting SAM to BAM</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort output.bam output.sorted <span class="co">#sorting the BAM reads</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index output.sorted.bam <span class="co">#indexing the sorted BAM</span></span></code></pre></div>
<li>
OR, pipe some of the commands in a one-liner:
</li>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">bwa</span> mem ref.fa read1.fq read2.fq <span class="kw">|</span> <span class="ex">samtools</span> view <span class="at">-S</span> <span class="at">-b</span> <span class="at">-</span> <span class="kw">|</span> <span class="ex">samtools</span> sort <span class="at">-o</span> output.sorted.bam <span class="at">-</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index output.sorted.bam</span></code></pre></div>
<li>
Blah blah.
</li>
<li>
You should be able to see the resultant files <code>file1</code> and <code>file2</code>.
</li>
</ul></details>
</div>
</div>
<div id="task-2-calling-snvs-using-strelka2" class="section level3">
<h3 class="hasAnchor">
<a href="#task-2-calling-snvs-using-strelka2" class="anchor"></a>Task 2: calling SNVs using Strelka2</h3>
<p>Now that we have the sequencing reads in order, we will start calling SNVs. Blah blah…</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#strelka example command</span></span></code></pre></div>
</div>
<div id="task-3-investigating-the-driver-gene" class="section level3">
<h3 class="hasAnchor">
<a href="#task-3-investigating-the-driver-gene" class="anchor"></a>Task 3: investigating the driver gene</h3>
<p>Text text</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#install.packages()</span></code></pre></div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:dong.rn@wehi.edu.au" class="email">dong.rn@wehi.edu.au</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:fu.j@wehi.edu.au" class="email">fu.j@wehi.edu.au</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p><a href="mailto:cameron.d@wehi.edu.au" class="email">cameron.d@wehi.edu.au</a><a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>Craig, D., Nasser, S., Corbett, R. et al. A somatic reference standard for cancer genome sequencing. Sci Rep 6, 24607 (2016). <a href="https://doi.org/10.1038/srep24607" class="uri">https://doi.org/10.1038/srep24607</a><a href="#fnref4" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniel Cameron, Ruining Dong, James Fu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
